ssg/build_yaml.py

```diff
@@ -51,53 +46,16 @@
 from .shims import unicode_func
 from .build_cpe import ProductCPEs
 import ssg.build_stig
-
-def dump_yaml_preferably_in_original_order(dictionary, file_object):
-    try:
-        return yaml.dump(dictionary, file_object, indent=4, sort_keys=False)
-    except TypeError as exc:
-        # Older versions of libyaml don't understand the sort_keys kwarg
-        if "sort_keys" not in str(exc):
-            raise exc
-        return yaml.dump(dictionary, file_object, indent=4)
-
-
-def add_sub_element(parent, tag, ns, data):
-    """
-    Creates a new child element under parent with tag tag, and sets
-    data as the content under the tag. In particular, data is a string
-    to be parsed as an XML tree, allowing sub-elements of children to be
-    added.
-
-    If data should not be parsed as an XML tree, either escape the contents
-    before passing into this function, or use ElementTree.SubElement().
-
-    Returns the newly created subelement of type tag.
-    """
-    namespaced_data = add_xhtml_namespace(data)
-    # This is used because our YAML data contain XML and XHTML elements
-    # ET.SubElement() escapes the < > characters by &lt; and &gt;
-    # and therefore it does not add child elements
-    # we need to do a hack instead
-    # TODO: Remove this function after we move to Markdown everywhere in SSG
-    ustr = unicode_func('<{0} xmlns="{3}" xmlns:xhtml="{2}">{1}</{0}>').format(
-        tag, namespaced_data, xhtml_namespace, ns)
-
-    try:
-        element = ET.fromstring(ustr.encode("utf-8"))
-    except Exception:
-        msg = ("Error adding subelement to an element '{0}' from string: '{1}'"
-               .format(parent.tag, ustr))
-        raise RuntimeError(msg)
-
-    # Apart from HTML and XML elements the rule descriptions and similar
-    # also contain <xccdf:sub> elements, where we need to add the prefix
-    # to create a full reference.
-    for x in element.findall(".//{%s}sub" % XCCDF12_NS):
-        x.set("idref", OSCAP_VALUE + x.get("idref"))
-        x.set("use", "legacy")
-    parent.append(element)
-    return element
+from .entities.common import (
+    XCCDFEntity,
+    SelectionHandler,
+    add_sub_element,
+    derive_id_from_file_name
+)
+from .entities.profile import (
+    Profile,
+    ProfileWithInlinePolicies,
```

Comment:
Only `ProfileWithInlinePolicies` is used in this file.

Reply:
Profile is used by other scripts (generating tables), so I would keep it here, as `build_yaml` could stay a "hub", where things are available for imports, but are defined elsewhere.